Inclusion of a Tutorial on an Idealized Case & Parameter Description of Feature Detection

fsenf opened the pull request:
## Description
- an additional tutorial notebook on tracking of a 2d test blob
- two additional notebooks on the effect of feature detection parameters

Work mainly contributed by @snilsn and reviewed and updated by me.

## Checklist
* [x] Have you followed our guidelines in Section "Contributing" from README.md? 
* [x] Have you checked already existing notebooks to get an idea about our tutorial style?
* [x] Have you extensively used markdown elements to structure and explain your notebook?
* [x] Have you self-reviewed your code / notebook and corrected any misspellings?

* [ ] Have you uploaded your example data to [zenodo]( https://zenodo.org/ ) (see README) if needed?
   * does not apply
* [x] Have you checked if your notebook is running in the recommended standard conda enviroment [see here]((docs/RunningTutorials.md) and updates the [list of required python packages](./requirements.txt) if needed?
* [x] Have you carefully checked that sphinx-based rendering is still working correctly?
* [ ] Have you written descriptive commit messages? 
  * I am always bad on this aspect :-)
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes been merged into this branch?

snilsn committed:
Adding notebooks of the basics (#3)

Added three additional tutorial notebooks for educational purposes:
* idealized cases without data input (using test data from CI testing routines)
* showing the standard workflow once, and then explain several parameters used in feature detection

Co-authored-by: Nils Pfeifer <nilspfeifer@protonmail.com>

fsenf committed:
revision of the tets blob notebook

fsenf committed:
Merge branch 'master' into idealized-cases

fsenf committed:
Merge branch 'master' of github.com:tobac-project/tobac-tutorials into idealized-cases

fsenf committed:
deleted explicit sphinx version in requirements

fsenf committed:
added 1st part on feature detection

fsenf committed:
updated size and order of blob plots

fsenf committed:
added two basic tutorials in the sidebar

fsenf committed:
revised feature detect. notebook, part 2

JuliaKukulies commented on index.rst:
```diff
@@ -27,7 +39,7 @@ Welcome to a Collection of Tutorials for the Python Package `tobac`
    :maxdepth: 1 
    :hidden:
-   :caption: Tracking in Simulaitons
+   :caption: Advanced Tracking: Simulaitons
```
```suggestion
   :caption: Advanced Tracking: Simulations
```

JuliaKukulies commented on requirements.txt:
```diff
@@ -1,4 +1,4 @@
-sphinx==3.5.4
+sphinx
```
Does the version not play any role anymore or why did you delete this?

fsenf replied:
Yeah, the problem comes with older sphinx versions and their dependencies to other packages. With the fixed version v3.5.4 e.g., I get the following error after make html

...
ImportError: cannot import name 'environmentfilter' from 'jinja2' 

Using the newest sphinx version helps and no import errors appear.

I remember that we needed to fix the version (some time ago) because the at-this-time newest sphinx version had some small bugs.

PS: sorry for double posting ...

JuliaKukulies commented on tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb:
```diff
@@ -0,0 +1,3354 @@
+   "id": "be1be87f",
+   "metadata": {},
+   "source": [
+    "This tutorial shows the most important step of tracking with `tobac` using a idealized case:\n",
```
```suggestion
    "This tutorial shows the most important step of tracking with `tobac` using an idealized case:\n",
```

JuliaKukulies commented on tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb:
```diff
@@ -0,0 +1,3354 @@
+   "outputs": [
+    {
+     "data": {
+      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAuEAAAO5CAYAAAC36vu8AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAAsTAAALEwEAmpwYAAEAAElEQVR4nOzdeZxkVX3//9e7l5lh2EZAjeACRg0iGDEaNcYA4hKJomLQGBRREzVqNlHcWDUuUfFrNGpEEdx+LiQMSEQWBRSTaCQubEqUzYAosswMMMxMd9fn98e9DTV1P7e61u6u7veTx300fercc0/dqu45fepzzkcRgZmZmZmZzZ+xhe6AmZmZmdly40G4mZmZmdk88yDczMzMzGy ...
```
```suggestion
    "Another step after the feature detection and tracking is the segmentation of the data. This means we find the area surrounding our features belonging to the same cluster. Logically, we now need the already detected features as an additional input."
```

fsenf replied:
This comment refers to the image, right? Don't how to deal with it ...

JuliaKukulies replied:
I am sorry, I have no idea why this image thing appears here. This was meant to be a direct suggestion for a super tiny typo: allready -> already, but I have realized that making direct changes in the notebook file somehow results in this when there is an image in the previous line.

fsenf replied:
OK, yes ... it is always a pain to review notebooks ... I corrected the spelling error. Update comes with the next commit ...

JuliaKukulies commented on tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb:
```diff
@@ -0,0 +1,3354 @@
+   "outputs": [
+    {
+     "data": {
+      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAuEAAAO5CAYAAAC36vu8AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAAsTAAALEwEAmpwYAAEAAElEQVR4nOzdeZxkVX3//9e7l5lh2EZAjeACRg0iGDEaNcYA4hKJomLQGBRREzVqNlHcWDUuUfFrNGpEEdx+LiQMSEQWBRSTaCQubEqUzYAosswMMMxMd9fn98e9DTV1P7e61u6u7veTx300fercc0/dqu45fepzzkcRgZmZmZmZzZ+xhe6AmZmZmdly40G4mZmZmdk88yDczMzMzGy ...
```
```suggestion
    "Another interesting property for real data are the lifetimes of our features. Tobac can also produce a histogram of this:"
```

fsenf replied:
and the same: This comment refers to the image, right? Don't how to deal with it ...

JuliaKukulies replied:
Same here, the only change I wanted to suggest was histgram -> histogram in l. 3297

fsenf replied:
Thanks for careful reading! This is also corrected in the next commit.

JuliaKukulies commented on tutorials/Basics/Methods-and-Parameters-for-Feature-Detection_Part_2.ipynb:
```diff
@@ -0,0 +1,1522 @@
+   "metadata": {},
+   "source": [
+    "This parameter can be understood best, when looking at real instead of snythethic data. An example of usage is given [here](../Example_vorticity_tracking_model/Example_vorticity_tracking_model.ipynb)"
+   ]
```
Maybe add this small section in `Methods-and-Parameters-for-Feature-Detection_Part_1` instead? It does probably not matter, but I felt it is more logically to mention the ability of wavelength filter together with the gaussian filter, because both input parameters apply some kind of filter on the input data.

JuliaKukulies commented on tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb:
```diff
@@ -0,0 +1,3354 @@
+   "outputs": [
+    {
+     "data": {
+      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAuEAAAO5CAYAAAC36vu8AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAAsTAAALEwEAmpwYAAEAAElEQVR4nOzdeZxkVX3//9e7l5lh2EZAjeACRg0iGDEaNcYA4hKJomLQGBRREzVqNlHcWDUuUfFrNGpEEdx+LiQMSEQWBRSTaCQubEqUzYAosswMMMxMd9fn98e9DTV1P7e61u6u7veTx300fercc0/dqu45fepzzkcRgZmZmZmZzZ+xhe6AmZmZmdly40G4mZmZmdk88yDczMzMzGy ...
```
_Just a small thought here:_ In all other example notebooks we introduce the different steps as 1.feature detection, 2.segmentation and 3. tracking. I know that the order actually does not matter since the segmentation is based on the feature detection output, so maybe we could clarify this here for users that the segmentation can be done before or after the linking. 

Also, one could actually use the output of the segmentation (`features_test` in your case) as input for the tracking with the advantage that information on the area (`ncells`) is added in the dataframe. One could also use the output of the tracking in the segmentation (`trajectories` in your case) with the advantage that  `mask` will contain only the features that are also linked to trajectories. Do you think these possibilities are worth mentioning here?

fsenf replied:
@JuliaKukulies : good point! I actually reordered Nils notebook such that segmentation comes later. The second segmentation is actually more an analysis step then something else. It does not impact tracking.

However, I did remember it wrong that the explanation in https://tobac.readthedocs.io/en/latest/ was also structured in this way. So now, I do have a tendency to shift it back again and to bring the second segmentation directly after the feature detection.

fsenf replied:
I left it as it is ... but included some of your arguments, okay?

JuliaKukulies replied:
Yes, OK!

JuliaKukulies reviewed (commented):
These three additional notebooks are VERY helpful and a great contribution in my opinion! Thanks a lot for this PR! I have only some small suggestions/questions and a few typo fixes before this can be merged.

JuliaKukulies commented on tutorials/Basics/Idealized-Case-1_Tracking-of-a-Test-Blob-in-2D.ipynb:
```diff
@@ -0,0 +1,3354 @@
+   "outputs": [],
+   "source": [
+    "dxy, dt = tobac.utils.get_spacings(test_data)"
+   ]
```
should be `tobac.utils.general.get_spacings` for `v2.0-dev`

fsenf committed:
spelling correction in index

fsenf committed:
included seaborn as conda package to be installed for tutorials

fsenf committed:
Notebook Update after Review:
- spelling correction in 2d Blob tracking
- included discussion of segmentation as part of analysis
- moved spectral band-pass filter to part 1

fsenf commented:
@JuliaKukulies : The update should address most of your comments.

fsenf committed:
Spelling correction in notebook for idealized tracking of a 2D bolb

fsenf commented:
@JuliaKukulies : all suggested correction should be included now! What do you think: Ready to merge?

JuliaKukulies reviewed (approved):
Great, thanks! Yes, please go ahead and merge!
